### ISSUE TITLE
[Enhancement] Prefer to keep entire statement intact when wrapping closure

### ISSUE THREAD

**Issue by ChristopherRogers** [nicklockwood/SwiftFormat#510]
I noticed some closures formatted oddly like the example below:

```swift
// Before
let keys = Array(ids!.keys).sorted { $0.localizedStandardCompare($1) == .orderedAscending }

// After
let keys = Array(ids!.keys).sorted { $0
    .localizedStandardCompare($1) == .orderedAscending
}
```

I thought that keeping `$0` on the same line would be more readable, since it wasn't going over the line limit anyway:

```swift
let keys = Array(ids!.keys).sorted {
    $0.localizedStandardCompare($1) == .orderedAscending
}
```

**Comment by nicklockwood** [comment]
Makes sense. I'll try to tweak the wrapping priorities to help with this case.

**Comment by nicklockwood** [comment]
@ChristopherRogers fixed in 0.43.3.
